eatin-client/src/components/accounts/profile.tsx

```diff
@@ -13,27 +13,28 @@ import TabList from "@mui/joy/TabList";
 import Tab from "@mui/joy/Tab";
 import TabPanel from "@mui/joy/TabPanel";
 import { RecipesCatalog } from "../ui/RecipesCatalog";
-import { useGetSavedRecipes } from "../functions/useGetSavedRecipes";
+import { useGetRecipesConnectionIsSaved } from "../../graphql/queries/recipes_connection_is_saved.query";
 
 export const Profile: FC = () => {
     const [categoryFilter, setCategoryFilter] = useState("");
     const [cuisineFilter, setCuisineFilter] = useState("");
     const { currentUser } = useAuth();
     const { data, error, loading } = useGetUserByIdQuery({
-        variables: { id: !!currentUser?.uid ? currentUser?.uid : "" },
+        variables: { id: currentUser ? currentUser.uid : "" },
```

Comment:
In this file and the rest files where you get some parameter of `currentUser`, add a `useMemo` hook that will calculate the parameter once. like this: \
`const kaki = useMeme(() => currentUser ? currentUser.uid : "", [currentUser])`